test-suite/forwardoption.cpp

```diff
@@ -576,135 +576,233 @@ void ForwardOptionTest::testMCPrices() {
                            analyticPrice, mcPrice, error, tolerance);
       }
    }
-
 }
 
 
-
 void ForwardOptionTest::testHestonMCPrices() {
    BOOST_TEST_MESSAGE("Testing forward option Heston MC prices...");
 
-   Real tolerance = 1e-4;
+   Option::Type optionTypes[] = { Option::Call, Option::Put };
 
-   Size timeSteps = 50;
-   Size numberOfSamples = 131072;
-   Size mcSeed = 42;
+   for (Size j=0; j<2; j++) {
 
-   Real q = 0.04;
-   Real r = 0.01;
-   Real sigma_bs = 0.245;
-   Real s = 100;
+      Option::Type type = optionTypes[j];
 
-   // Test 1: Set up an equivalent flat Heston and compare to analytical BS pricing
-   Real v0 = sigma_bs * sigma_bs;
-   Real kappa = 1e-8;
-   Real theta = 0.08;
-   Real sigma = 1e-8;
-   Real rho = -0.93;
+      Real mcTolerance = 5e-4;
+      Real analyticTolerance = 5e-4;
 
-   DayCounter dc = Actual360();
-   SavedSettings backup;
-   Date today = Date::todaysDate();
-   Settings::instance().evaluationDate() = today;
+      Size timeSteps = 50;
+      Size numberOfSamples = 32768;
+      Size mcSeed = 42;
 
-   Date exDate = today + 1*Years;
-   ext::shared_ptr<Exercise> exercise(
+      Real q = 0.04;
+      Real r = 0.01;
+      Real sigma_bs = 0.245;
+      Real s = 100;
+
+      // Test 1: Set up an equivalent flat Heston and compare to analytical BS pricing
+      Real v0 = sigma_bs * sigma_bs;
+      Real kappa = 1e-8;
+      Real theta = sigma_bs * sigma_bs;
+      Real sigma = 1e-8;
+      Real rho = -0.93;
+
+      DayCounter dc = Actual360();
+      SavedSettings backup;
+      Date today = Date::todaysDate();
```

Comment:
The whole test suite sets a fixed evaluation date before starting, so I suggest using
```
      Date today = Settings::instance().evaluationDate();
```
instead.

The idea is that having the evaluation date dependent on the actual today's date makes it prone to spurious errors, upon which we have to increase the tolerance to pass.  Using a variable date _might_ find some actual errors, but it's unlikely and we don't run the test suite every day anyway.

(Using the fixed date might also allow you to get the tolerance back to its original value in the previous test case.)

Reply:
This certainly seems sensible. The other tests in that test suite use the current day's date as well (I'd borrowed their convention) - shall I make the change across the suite while I'm doing it?